### PR TITLE
Zip handling

### PR DESCRIPTION
**Pull request by Vilsol**

### CODE REVIEW

File: SatisfactoryModLoader/mod/Coremods.cpp [hunk 1]
```diff
@@ -42,5 +40,27 @@ namespace SML {
...
+
+			// TODO Support ZIP loaded memory mods
+			/*
+			if (MemoryGetProcAddress(hLib, "DELAYED_LOAD")) {
+				Utility::warning("Warning: Coremod loaded -> ", entry.path().filename(), " (DELAYED_LOAD). Please remove coremods before submitting bug reports!");
+				MemoryFreeLibrary(hLib);
+				delayedMemoryCoremods.push_back(hLib);
+				return;
+			}
+			*/
```
Comment (SuperCoder7979) [inline SatisfactoryModLoader/mod/Coremods.cpp]: Delete this please

File: SatisfactoryModLoader/mod/ModHandler.cpp [hunk 2]
```diff
@@ -227,6 +332,25 @@ namespace SML {
...
+								Utility::error("Failed loading object: ", objPath, " in ", archive);
+								continue;
+							}
+
+							void* result = (unsigned char *)malloc(obj->size());
+							size_t bytes = obj->read(result, obj->size());
+							obj->close();
+
+							loadMemoryCoreMod(result, obj->size(), objPath)
+							*/
```
Comment (SuperCoder7979) [inline SatisfactoryModLoader/mod/ModHandler.cpp]: This as well

File: SatisfactoryModLoader/mod/ModHandler.cpp [hunk 3]
```diff
@@ -15,15 +18,110 @@
...
+					catch (...) {}
+
+					if (!dataJson.contains("objects")) {
+						Utility::error("data.json missing objects in ", archive);
+						continue;
+					}
+
+					auto objects = dataJson["objects"];
+
+					for (auto it = objects.begin(); it != objects.end(); ++it) {
```
Comment (SuperCoder7979) [inline SatisfactoryModLoader/mod/ModHandler.cpp]: Is it possible to infer what the objects are in the zip if an old style data.json exists?

File: SatisfactoryModLoader/util/EnvironmentValidity.cpp [hunk 4]
```diff
@@ -8,28 +8,97 @@
...
+			}
+		}
+
+		void createDirectory(std::string path, bool hidden) {
+			CreateDirectoryA(path.c_str(), NULL);
+
+			if (hidden) {
+				SetFileAttributesA(path.c_str(), FILE_ATTRIBUTE_HIDDEN);
+			}
+		}
```
Comment (SuperCoder7979) [inline SatisfactoryModLoader/util/EnvironmentValidity.cpp]: All of this needs to be in Utility.cpp, not EnvironmentValidity.cpp

File: SatisfactoryModLoader/util/EnvironmentValidity.h [hunk 5]
```diff
@@ -3,6 +3,33 @@
...
+
+		// Cleanup all hidden directories (hiddenmod, hiddencoremod)
+		void cleanupHiddenDirectories();
+
+		// Delete directory recursively
+		void deleteDirectoryRecursive(std::string path);
+
+		// Create an optionally hidden directory
+		void createDirectory(std::string path, bool hidden);
+
```
Comment (SuperCoder7979) [inline SatisfactoryModLoader/util/EnvironmentValidity.h]: Ditto here

File: SatisfactoryModLoader/mod/ModHandler.cpp (hunk 3 above)
Comment (Vilsol) [reply SatisfactoryModLoader/mod/ModHandler.cpp]: Yes, but in general very bad idea

**Pull request closed without merging**